itests/util/src/main/java/org/apache/hive/jdbc/miniHS2/MiniHS2.java

```diff
@@ -481,6 +488,23 @@ private String getZKBaseJdbcURL() throws Exception {
     throw new Exception("Server's HiveConf is null. Unable to read ZooKeeper configs.");
   }
 
+  /**
+   * return HTTP connection URL for this server instance
+   * @return
+   * @throws Exception
+   */
+  public String getHttpJdbcURL() throws Exception {
+    String transportMode = getConfProperty(ConfVars.HIVE_SERVER2_TRANSPORT_MODE.varname);
+    if(!transportMode.equalsIgnoreCase(HS2_ALL_MODE)) {
+      return getJdbcURL();
+    }
+    //CAUTION: This will result in concurrency issues.
```

Comment:
Is this the reason we disabled the parallel attribute in the pom.xml file?

Reply:
yes